vince/views.py

```diff
@@ -11722,34 +11899,86 @@ def form_valid(self, form):
                     logger.debug(f"FOUND RESERVATION for {cve.cve_name}")
                     cve_res.cve_info = cve
                     cve_res.save()
-            
+
         cve.cwe = json.dumps(cwes)
         #if was:
         #    cve.work_around = json.dumps(was)
         cve.references = json.dumps(refs)
         cve.save()
 
         prods = []
+        prods2 = []
         if cveprodformset.is_valid():
+            prodchanged = False
+
             for f in cveprodformset:
                 cd = f.cleaned_data
-                prods.append(CVEAffectedProduct(cve = cve,
-                                                name=cd['name'],
-                                                version_affected=cd['version_affected'],
-                                                version_name=cd['version_name'],
-                                                version_value=cd['version_value'],
-                                                organization=cd['organization']))
+                if cd['organization']:
+                    contact = Contact.objects.filter(id=int(cd['organization']))
+                    prods.append(CVEAffectedProduct(cve = cve,
+                                                    name=cd['cve_affected_product'],
+                                                    version_affected=cd['version_affected'],
+                                                    version_name=cd['version_name'],
+                                                    version_value=cd['version_value'],
+                                                    organization=contact[0].vendor_name))
+
+                    #check if vendorProduct exists, if not create it.
+                    checkprod = VendorProduct.objects.filter(name = cd['cve_affected_product'])
```

Comment:
Shouldn't this check for organization match as well? like below 

``` checkprod = VendorProduct.objects.filter(name = cd['cve_affected_product'], organization=contact[0])```

Preferably
``` checkprod = VendorProduct.objects.filter(name__iexact = cd['cve_affected_product'], organization=contact[0])```

Reply:
Few bugs found when running further Test Cases against VendorProduct with multi vendors. A duplicate product name under two vendors causes Null-Property reference exception. On delete the cascade will also fail.

Reply:
> Shouldn't this check for organization match as well? like below
> 
> ` checkprod = VendorProduct.objects.filter(name = cd['cve_affected_product'], organization=contact[0])`
> 
> Preferably ` checkprod = VendorProduct.objects.filter(name__iexact = cd['cve_affected_product'], organization=contact[0])`

Yes, it should be!

Reply:
> Few bugs found when running further Test Cases against VendorProduct with multi vendors. A duplicate product name under two vendors causes Null-Property reference exception. On delete the cascade will also fail.

I guess it makes sense why it is failing, because there is a single VendorProduct table with reference to contact_fk and no uniqueness is built into that table property ['name']
I wonder how to solve this? Accessing UUID instead of name fields? Prepending the contact['name'] to a vendorproduct['name'] so ACME_Product1 when saving?

Then when displaying or modifying the name break the string on contact['name']_ ?

Or add a different another property as this above and keeping the ['name'] field?

Based on your feedback, I can spend some time working on a patch tomorrow for both this what I can come up with for org_auto above.

Reply:
@z-priest  Don't bother more development, Part of the testCases suggest some patches. I will apply and run the test cases again and push updates later. This way we can avoid duplicate work as it is hard to push test outputs to you in our current setup.